scripts/measures/2018/get-strata-uuids-from-ecqm-zip-2018.js

```diff
@@ -0,0 +1,171 @@
+  components.forEach((component, index) => {
+    const ids = component.populationCriteriaSection[0].component;
+    const eMeasureUuids = {
+      initialPopulationUuid: ids.find(item => item.initialPopulationCriteria).initialPopulationCriteria[0].id[0].$.root,
+      denominatorUuid: ids.find(item => item.denominatorCriteria).denominatorCriteria[0].id[0].$.root,
+      numeratorUuid: ids.find(item => item.numeratorCriteria).numeratorCriteria[0].id[0].$.root
+    };
+
+    const denominatorException = ids.find(item => item.denominatorExceptionCriteria);
+    if (denominatorException) {
+      eMeasureUuids.denominatorExceptionUuid = denominatorException.denominatorExceptionCriteria[0].id[0].$.root;
+    }
+
+    const denominatorExclusion = ids.find(item => item.denominatorExclusionCriteria);
+    if (denominatorExclusion) {
+      eMeasureUuids.denominatorExclusionUuid = denominatorExclusion.denominatorExclusionCriteria[0].id[0].$.root;
+    }
+    strata[index].eMeasureUuids = eMeasureUuids;
+  });
+
+  return strata;
+}
+
+// gather list of xml files
+rimraf.sync(tmpDir);
+new AdmZip(zipPath).extractAllTo(tmpDir, true);
+// each measure has its own zip, collect name of SimpleXML files
+const xmlFiles = fs.readdirSync(tmpDir).map(measureZip => {
+  const zip = new AdmZip(path.join(tmpDir, measureZip));
+
+  const filename = zip.getEntries()
+    .find(entry => entry.entryName.match(/[0-9]\.xml$/))
+    .entryName;
+
+  // extract 'CMS75v5.xml' to /xmls
+  zip.extractEntryTo(filename, tmpDir + '/xmls', false, true);
+
+  return filename.split('/')[1];
+});
+
+// parse files into JavaScript objects
+const promisifiedParseString = Promise.promisify(parseString);
+Promise.all(
+  xmlFiles.map(xmlFile => {
+    return promisifiedParseString(fs.readFileSync(path.join(tmpDir, '/xmls', xmlFile)));
+  })
+)
+// extract data from converted JavaScript objects
+  .then(docs => {
+    return _.compact(docs.map(doc => {
+      const measure = doc.QualityMeasureDocument;
+      const emeasureid = measure.subjectOf[0].measureAttribute[0].value[0].$.value;
+      if (emeasureid === '138') {
+        console.warn('WARNING: CMS138v6 has one numerator but multiple populations and needs to be added manually - see /tmp/ecqm/EP_EC_CMS138v6_NQF0028_PREV_Tobacco.zip');
+        return;
+      }
+      if (emeasureid === '145') {
+        console.warn('WARNING: CMS145v6 has one numerator but two initial populations and needs to be added manually - see /tmp/ecqm/EC_CMS145v6_NQFXXXX_CAD_BB.zip');
+        return;
+      }
+      if (emeasureid === '160') {
+        console.warn('WARNING: CMS160v6 has one numerator but two initial populations and needs to be added manually - see /tmp/ecqm/EC_CMS160v6_NQF0712_Dep_PHQ9.zip');
+        return;
+      }
+      if (emeasureid === '347') {
+        console.warn('WARNING: CMS347v1 has one numerator but three initial populations and needs to be added manually - see /tmp/ecqm/EP_EC_CMS347v1_NQFXXXX_Statin_Therapy_Prev.zip');
```

Comment:
@saquino0827 do these edge cases still exist? i wrote the first version of this script for 2017 measures, wondering if the 2018 ecqm zips are the same

Reply:
If I understand it, they do exist. I thought that if they had multiple strata(numerator) values, then they could not be extracted from the initial script and needed to be handled separately. Whenever I ran the script for the current zip file, the 4 eMeasureId's would always fail for the same reason.